packages/remix/src/build.ts

```diff
@@ -104,15 +104,17 @@ export const build: BuildV2 = async ({
     meta
   );
 
-  const { cliType, packageJsonPath, lockfileVersion } = await scanParentDirs(
-    entrypointFsDirname
-  );
+  const { cliType, packageJsonPath, lockfileVersion, lockfilePath } =
+    await scanParentDirs(entrypointFsDirname);
 
   if (!packageJsonPath) {
     throw new Error('Failed to locate `package.json` file in your project');
   }
 
-  const pkgRaw = await fs.readFile(packageJsonPath, 'utf8');
+  const [lockfileRaw, pkgRaw] = await Promise.all([
+    lockfilePath ? fs.readFile(lockfilePath) : null,
```

Comment:
Lock files can get pretty big. Maybe copy the file instead of keeping it in memory?

Reply:
Perhaps. I think we should wait and see if it would cause an issue for anyone. Machines tend to have a decent amount of memory these days.